kategory-docs/docs/typeclasses/applicative/README.md

```diff
@@ -0,0 +1,140 @@
+
+In the following example we will define 3 invocations that may as well be remote or local services each one of them returning different results in the same computational context of `Option`
+
+```kotlin:ank
+import kategory.*
+
+fun profileService(): Option<String> = Option("Alfredo Lambda")
+fun phoneService(): Option<Int> = Option(55555555)
+fun addressService(): Option<List<String>> = Option(listOf("1 Main Street", "11130", "NYC"))
+```
+
+This more or less illustrate the common use case of performing several independent operations where we need to get all the results together
+
+Kategory features an [Applicative Builder](/docs/patterns/applicativebuilder) that allows you to easily combine all the independent operations into one result.
+
+```kotlin:ank
+data class Profile(val name: String, val phone: Int, val address: List<String>)
+
+val r: Option<Tuple3<String, Int, List<String>>> = Option.tupled(profileService(), phoneService(), addressService())
+r.map { Profile(it.a, it.b, it.c) } 
+```
+
+The Applicative Builder also provides a `map` operations that is able to abstract over arity in the same way as `tupled`
+
+```kotlin:ank
+Option.map(profileService(), phoneService(), addressService(), { name, phone, address ->
+  Profile(name, phone, address)
+})
+```
+
+### Main Combinators
+
+#### pure
+
+Lifts a value into the computational context of a type constructor
+
+`fun <A> pure(a: A): HK<F, A>`
+
+```kotlin:ank
+Option.pure(1) // Option(1)
+```
+
+#### ap
+
+Apply a function inside the type constructor's context
+
+`fun <A, B> ap(fa: HK<F, A>, ff: HK<F, (A) -> B>): HK<F, B>`
+
+```kotlin:ank
+Option.ap(Option(1), Option({ n: Int -> n + 1 })) // Option(2)
+```
+
+#### Other combinators
+
+For a full list of other useful combinators available in `Applicative` see the [`KDoc](/kdocs/typeclasses/applicative)
+
+### Syntax
+
+#### HK<F, A>#pure
+
+Lift a value into the computational context of a type constructor
+
+```kotlin:ank
+1.pure<TryHK, Int> // Try.Success(1)
+```
+
+#### HK<F, A>#ap
+
+Apply a function inside the type constructor's context
+
+```kotlin:ank
+Option(1).ap(Option({ n: Int -> n + 1 })) // Option(2)
+```
+
+#### HK<F, A>#map2
+
+Map 2 values inside the type constructor context and apply a function to their cartesian product
+
+```kotlin:ank
+Option(1).map2(Option("x"), { z: Tuple2<Int, String> ->  z.a + z.b }) // Option("1x")
+```
+
+#### HK<F, A>#map2Eval
+
+Lazily map 2 values inside the type constructor context and apply a function to their cartesian product.
+Computation happens when `.value()` is invoked.
+
+```kotlin:ank
+Option(1).map2(Eval { Option("x") }, { z: Tuple2<Int, String> ->  z.a + z.b }).value() // Option("1x")
+```
+
+
+### Laws
+
+Kategory provides [`ApplicativeLaws`](/docs/typeclasses/laws#applicativelaws) for internal verification of lawful instances and third party apps creating their own Applicative instances.
```

Comment:
We should flesh out the explanation for laws, or at least use the term test or test suite to refer to them, as that relates more to coding concepts.